components/keys/src/types.rs

```diff
@@ -211,14 +212,108 @@ impl Display for Key {
     }
 }
 
+#[derive(Clone, Copy, Debug, Default, Eq, PartialEq, Ord, PartialOrd, Hash)]
+pub struct TimeStamp(u64);
+
+const TSO_PHYSICAL_SHIFT_BITS: u64 = 18;
+
+impl TimeStamp {
+    /// Create a time stamp from physical and logical components.
+    pub fn compose(physical: u64, logical: u64) -> TimeStamp {
+        TimeStamp((physical << TSO_PHYSICAL_SHIFT_BITS) + logical)
+    }
+
+    pub fn now_sec() -> TimeStamp {
```

Comment:
I don't think this method makes sense, since TimeStamp is a composition of the physical part and the logical part. This function should be only useful in vary rare and limited cases.

Reply:
It's not a new method, it used to be a function in one of the `util` functions which I just moved to being an inherent method,